### DISCUSSION TITLE
Server function in an npm package

### DISCUSSION THREAD

**Discussion by gquagliano** [vercel/next.js#56653] [Help]
### Summary

Hello! I'm trying to distribute a server function in an npm package. But it always runs on the client. Is this even possible?

### Additional information

Next 13.4.2

The idea is to distribute a function that I want to be automatically executed as a server action.

For example:

The npm package:
```js
async function myFunc(action) {
    "use server";
    await action();
}
module.exports = { myFunc };
```

The Next app, `app/page.js`:
```js
import { myFunc } from "my-package";
import { someAction } from "@/actions/test.js";
export default function Page() {
    async function test() {
        await myFunc(someAction);
    }

    return <button onClick={test}>Test</button>;
}
```

Results: `myFunc` runs on the client and `someAction` does run on the server.

Desired behavior: `myFunc` (from the library) runs on the server too.

**Comment by ajwootto** [comment]
I'm having the same problem, it seems like the webpack bundler seemingly ignores "use server" directives when present in a node_module package

**Comment by 0xbuidler** [comment]
@ajwootto @gquagliano 

I think you might be better off by not building the package but instead use `transpilePackages: ["@acme/myPackage"]` in your `next.config.mjs`. I'll try this tomorrow to see if it works as expected.

**Comment by gquagliano** [comment]
That may work for now (haven't tried it yet), but I think this is a valid use case. Thanks for the suggestion!

**Comment by ajwootto** [comment]
Not sure if you've tried recently but this should be fixed on Next 14.1 I think

**Comment by gquagliano** [comment]
It's true this is quite old already. I'm trying again with next 14.1.4 (which I love btw) and it still runs on the client, but I get the error `(in promise) TypeError: (0 , _module_name__WEBPACK_IMPORTED_MODULE_2__.myFunc) is not a function` (in the browser). If I inspect it it's actually undefined.

I had to move the module's `"use server";` to the top of the file.

It's a CJS module and I've installed it from a local folder, in case that matters.
